cdm/src/main/java/uk/ac/rdg/resc/edal/dataset/cdm/CdmGridDataSource.java

```diff
@@ -352,7 +354,8 @@ public WrappedArray(VariableDS var, Array arr, boolean needsEnhance, int[] shape
             this.rangesList = rangesList;
 
             if (needsEnhance && arr != null) {
-                this.arr = var.convertScaleOffsetMissing(arr);
+                this.arr = var.convertMissing(arr);
+                this.arr = var.applyScaleOffset(this.arr);
```

Comment:
You'll want to apply scale/offset **before** converting missing values. That's because, for example, values might not fall outside the valid range (which `convertMissing()` checks) until scale/offset have been applied. Instead, I would do:
```java
this.arr = var. convert(this.arr, false, true,  true);
```
That defers to [EnhanceScaleMissingUnsignedImpl.convert(Array, boolean, boolean, boolean)](https://github.com/Unidata/thredds/blob/5.0.0/cdm/src/main/java/ucar/nc2/dataset/EnhanceScaleMissingUnsignedImpl.java#L523), which will apply the enhancements in the proper order. It also avoids creating an unnecessary temporary Array copy.

Reply:
Thank you @cwardgar !